_stbt/core.py

```diff
@@ -1191,6 +1191,52 @@ def argparser():
     return parser
 
 
+class FrameObject(object):
+    def __init__(self, frame):
+        property_names = sorted([
+            p for p in dir(self.__class__)
+            if isinstance(getattr(self.__class__, p), property)])
+        if 'is_visible' not in property_names:
+            raise TypeError(
+                "Classes deriving from FrameObject must define a property "
+                "named 'is_visible'")
+        self.__attrs = ["is_visible"] + sorted(
+            x for x in property_names
+            if x != "is_visible" and not x.startswith('_'))
+
+        if frame is None:
+            raise ValueError("FrameObject: frame must not be None")
+        self._frame = frame
+
+    def __repr__(self):
+        if self:
+            args = ", ".join(
+                "%s=%r" % (x, getattr(self, x)) for x in self.__attrs)
+        else:
+            args = "%s=False" % self.__attrs[0]
+
+        return "%s(%s)" % (self.__class__.__name__, args)
+
+    def __nonzero__(self):
+        return bool(getattr(self, self.__attrs[0]))
+
+    def _astuple(self):
+        if self:
+            return tuple(getattr(self, x) for x in self.__attrs)
+        else:
+            return (False,)
+
+    def __cmp__(self, other):
+        # pylint: disable=W0212
+        if isinstance(other, self.__class__):
+            return cmp(self._astuple(), other._astuple())
+        else:
+            return NotImplemented
+
+    def __hash__(self):
+        return hash(self._astuple())
```

Comment:
Why do we want these to be hashable? Why would you use them in a set, or as keys in a dict? If you had 2 different frames (but showing the same UI elements, at least those exposed by properties of the FrameObject) they'd be treated as the same key.

I understand why we want them to be `__eq__`, and given that "Hashable objects which compare equal must have the same hash value", then I suppose this can't hurt. Just wondering if you had a particular use-case in mind.